compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -684,7 +680,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         // that collisions are ok here and this shouldn't
         // really show up for end-user.
         let (str_name, kind) = match hir_name {
-            ParamName::Plain(ident) => (ident.name, hir::LifetimeParamKind::InBand),
+            ParamName::Plain(ident) => (ident.name, hir::LifetimeParamKind::Explicit),
```

Comment:
This is the only suspicious change.

Reply:
Async fns declare named elided lifetimes. We should give them a category here -- not sure if it's `LifetimeParamKind::Explicit` though. Maybe I can add `LifetimeParamKind::Explicit` back. 

It all funnels into `hir::LifetimeDefOrigin::ExplicitOrElided` though, so I don't think the distinction matters?

Reply:
This case should only happen when synthetizing lifetimes for an async fn return `impl Future` type. There should not be a difference, but we got bit before.

Reply:
Once `LifetimeDefOrigin` is removed, `LifetimeParamKind` is only used in lifetime resolution diagnostics and rustdoc to filter elided lifetimes. Therefore `LifetimeParamKind::Explicit` is fine.